entrypoint.sh

```diff
@@ -25,8 +25,14 @@ if [ "$(id -u)" = '0' ]; then
 	echo "Using the injected certificate/privatekey pair" 
     fi
     # Fixing the ownership and permissions
-    chown postgres:postgres "${PG_SERVER_KEY}" "${PG_SERVER_CERT}"
-    chmod 600 "${PG_SERVER_KEY}"
+    for file in "$PG_SERVER_KEY" "$PG_SERVER_CERT"; do
+        if [ $(stat -c %U:%G "$file" ) != "postgres:postgres" ]; then
+            chown postgres:postgres "$file"
+        fi
+        if [ $(stat -c %a "$file") != "600" ]; then
```

Comment:
600 or less, not just 600. It can be 400.
Check the group and world non-writable bit, and that the user's execution is not set.

Reply:
The original code only made sure that the permission was `600`, are you saying that we should implement extra functionality now?

Reply:
Nope, not what I'm saying.
However, your suggested `if-test` is fine.

Reply:
Actually, I retract that. If the permissions were 400 (for example), it's fine for postgres, but the entrypoint would now fail.